source/slang/slang-emit-wgsl.cpp

```diff
@@ -0,0 +1,968 @@
+#include "slang-emit-wgsl.h"
+
+// A note on row/column "terminology reversal".
+//
+// This is an "terminology reversing" implementation in the sense that
+// * "column" in Slang code maps to "row" in the generated WGSL code, and
+// * "row" in Slang code maps to "column" in the generated WGSL code.
+//
+// This means that matrices in Slang code end up getting translated to
+// matrices that actually represent the transpose of what the Slang matrix
+// represented.
+// Both API's adopt the standard matrix multiplication convention whereby the
+// column count of the matrix on the left hand side needs to match row count of
+// the matrix on the right hand side.
+// For these reasons, and due to the fact that (M_1 ... M_n)^T = M_n^T ... M_1^T,
+// the order of matrix (and vector-matrix products) products must also reversed
+// in the WGSL code.
+//
+// This may lead to confusion (which is why this note is referenced in several
+// places), but the benefit of doing this is that the generated WGSL code is
+// simpler to generate and should be faster to compile.
+// A "terminology preserving" implementation would have to generate lots of
+// 'transpose' calls, or else perform more complicated transformations that
+// end up duplicating expressions many times.
+
+namespace Slang {
+
+void WGSLSourceEmitter::emitSwitchCaseSelectorsImpl(
```

Comment:
you should be able to delete this now.

Reply:
There is still the issue that cases sharing a block have different syntax in WGSL, [as mentioned in the description of the commit that adds this hook to the C-like emitter](https://github.com/shader-slang/slang/pull/5006/commits/432a57ae3bd7af2d521beedc1f4aef8a814d58b6):



    The C-like emitter will emit code like this:
    
        switch(a.x)
        {
    
        case 0:
        case 1:
        {
           ...
        } break;
    
        ...
    
        }
    
    This is not allowed in WGSL. Instead, selectors for cases that share a body must [1] be
    separated by commas, like this:
    
        switch(a.x)
        {
    
        case 0, 1:
        {
           ...
        } break;
    
        ...
    
        }
    
    To prepare for addressing issue #4807, this patch makes the emission of switch case
    selectors overloadable.
    
    [1] https://www.w3.org/TR/WGSL/#syntax-case_selectors

Would you handle it some other way?

Reply:
I see, this makes sense then. Can you include this explanation in the comment?